src/include/migraphx/op/unique.hpp

```diff
@@ -0,0 +1,340 @@
+                else if(input_data[idx1] > input_data[idx2])
+                    return false;
+            }
+            return input_data[idx1] < input_data[idx2];
+        };
+        std::map<size_t, y_info, decltype(idx_less_fn)> uniq_val_map(idx_less_fn);
+
+        std::vector<std::size_t> x_rev_indices;
+
+        // go through all the elements and find the unique elements..
+        size_t count_x = input_data.size();
+        for(size_t f_idx = 0, x_idx = 0; f_idx < count_x; f_idx += chunk_sz, x_idx++)
+        {
+            y_info entry          = {.y_idx = uniq_val_map.size(), .x_idx = x_idx};
+            auto [itr, added_new] = uniq_val_map.insert({f_idx, entry});
+            itr->second.ct++;
+            x_rev_indices.push_back(itr->second.y_idx);
+        }
+
+        std::vector<std::size_t> y_indices(uniq_val_map.size());
+        std::vector<std::size_t> y_count(uniq_val_map.size());
+        std::vector<std::size_t> y2x_indices(uniq_val_map.size());
+        size_t idx = 0;
+        // the unique elements are now sorted:
+        // post-processing for all the return indices.
+        for(const auto& v : uniq_val_map)
+        {
+            y2x_indices[v.second.y_idx] = idx;
+            y_indices[idx]              = v.second.x_idx;
+            y_count[idx]                = v.second.ct;
+            idx++;
+        }
+        // update x_rev_indices as per the sorted order of y_indices
+        for(auto& i : x_rev_indices)
+            i = y2x_indices[i];
+
+        return std::tuple{y_indices, x_rev_indices, y_count};
+    }
+
+    // CASE UNSORTED:
+    //
+    // To process into an un-sorted unique series of elements/chunks:
+    // For chunk size = 1 is a simple element, else use a flat representation of a tensor obj
+    // Go through the input elements/chunks one by one with inline processing of indices..
+
+    // INPUT x: [2, 1, 1, 3, 4, 3], attr_sorted = 0;
+
+    // OUTPUT(s): indices..
+    // y_indices: [0, 1, 3, 4]  --- first incidence, in terms of index in sequence x
+    // x_rev_indices: [0, 1, 1, 2, 3, 2] --- x seen in terms of indices of unique sequence y
+    // y_count: [1, 2, 2, 1] -- count at each y_index. sum = len(x)
+
+    // NOTE: y [2, 1, 3, 4]   --- the unique output is constructed from x[y_indices[...]]
+    // Output data structures: y_indices, x_rev_indices, y_count are processed inline.
+
+    template <class T>
+    auto unsorted_uniq_indices(const T& input_data, size_t chunk_sz) const
+    {
+        auto idx_less_fn = [&](auto idx1, auto idx2) {
+            // for efficiency, loop is bypassed for single elements. ct = 1
+            for(size_t ct = chunk_sz; ct > 1; idx1++, idx2++, ct--)
+            {
+                if(input_data[idx1] < input_data[idx2])
+                    return true;
+                else if(input_data[idx1] > input_data[idx2])
+                    return false;
+            }
+            return input_data[idx1] < input_data[idx2];
+        };
```

Comment:
This can use `std::lexicographical_compare` as well, but instead of repeating this, I think you should make a function so it can be reused:

```cpp
template<class T>
auto make_idx_less_fn(const T& input_data, size_t chunk_sz) const
{
    assert(idx1+chunk_sz < input_data.size());
    assert(idx2+chunk_sz < input_data.size());
    return [&input_data, chunk_sz](auto idx1, auto idx2) {
        return std::lexicographical_compare(input_data.begin()+idx1,
                                            input_data.begin()+idx1+chunk_sz,
                                            input_data.begin()+idx2,
                                            input_data.begin()+idx2+chunk_sz);
    };
}
```

Reply:
Done.

Reply:
I dont see this change in the latest fixes you pushed.

Reply:
Sorry, I am referring to the use of lexicographic compare, and that is already added.

Reply:
Can you add this function though so you dont need to duplicate it twice?

Reply:
Done.